alexey.zhuravlov/task5/bank-service-api/src/main/java/com/hillel/servlets/AccountServlet.java

```diff
@@ -0,0 +1,144 @@
+    private Map<Integer, Account> accountMap = new ConcurrentHashMap<>();
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        resp.setContentType("application'json");
+        Gson gson = new Gson();
+        String accountId = req.getParameter("accountId");
+
+        if (accountId == null) {
+            resp.getOutputStream().println(gson.toJson(accountMap));
+        } else {
+            try {
+                if (accountMap.get(Integer.valueOf(accountId)) != null) {
+                    resp.getOutputStream().println(gson.toJson(accountMap.get(Integer.valueOf(accountId))));
+                } else {
+                    Status status = new Status(false, "Account not found");
+                    resp.getOutputStream().print(gson.toJson(status));
+                    resp.setStatus(404);
+                }
+            } catch (Exception e) {
+                log.error("Exception while tried to get account: ", e);
+                Status status = new Status(false, "Bad request");
+                resp.getOutputStream().print(gson.toJson(status));
+                resp.setStatus(400);
+            }
+        }
+        resp.getOutputStream().flush();
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws  IOException {
+        resp.setContentType("application/json");
+        Gson gson = new Gson();
+        Status status;
+
+        try {
+            StringBuilder sb = new StringBuilder();
+            String s;
+            while ((s = req.getReader().readLine()) != null) {
+                sb.append(s);
+            }
+
+            CreateAccountRequest createAccountRequest = gson.fromJson(sb.toString(), CreateAccountRequest.class);
+
+            Account account = new Account(currentAccountId, createAccountRequest.getBalance(), createAccountRequest.getUser());
+            accountMap.put(account.getId(), account);
+            currentAccountId++;
+            status = new Status(true, "Account was added");
+            resp.getOutputStream().print(gson.toJson(account));
+        } catch (Exception e) {
+            log.error("Exception while tried to add account: ", e);
+            status = new Status(false, "Failed to add account");
+            resp.setStatus(500);
+        }
+        resp.getOutputStream().print(gson.toJson(status));
+        resp.getOutputStream().flush();
+    }
+
+    @Override
+    protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        resp.setContentType("application/json");
+        Gson gson = new Gson();
+        Status status;
+
+        try {
+            StringBuilder sb = new StringBuilder();
+            String s;
+            while ((s = req.getReader().readLine()) != null) {
+                sb.append(s);
+            }
+
+            Account account = gson.fromJson(sb.toString(), Account.class);
```

Comment:
You can remove code with StringBuilder
and use  gson.fromJson(**request.getReader()**

This code was from old example...